File: lib/src/model/region_info.dart
part of dr_libphonenumber;

/// Check https://countrycode.org/ for detail.
class RegionInfo {
  /// The region code or calling code.
  final int regionCode;

  /// The phone number excluding the [regionCode].
  final int phoneNumberValue;

  /// The country code.
  final String? countryCode;

  /// The formatted phone number with combination of [regionCode] & [phoneNumberValue].
  final String? formattedPhoneNumber;

  RegionInfo({
    this.regionCode = 0,
    this.phoneNumberValue = 0,
    this.countryCode,
    this.formattedPhoneNumber,
  });

  @override
  String toString() {
    return '[RegionInfo prefix=$regionCode, phoneNumberValue=$phoneNumberValue, iso=$countryCode, formatted=$formattedPhoneNumber]';
  }
}

File: packages/dr_libphonenumber/rust/src/dr_libphonenumber/models/region_info.rs
use flutter_rust_bridge::frb;

/// Check https://countrycode.org/ for detail.
#[derive(Debug, PartialEq)]
#[frb(dart_metadata=("immutable" import "package:meta/meta.dart" as meta))]
pub struct DrRegionInfo {
    /// The region code or calling code.
    pub region_code: u16,

    /// The phone number excluding the [regionCode].
    pub phone_number_value: u64,

    /// The country code.
    pub country_code: String,

    /// The formatted phone number with combination of [regionCode] & [phoneNumberValue].
    pub formatted_phone_number: String,
}
